benchmarks/Csharp/Benchmarks/Rich/Scheduler.cs

```diff
@@ -0,0 +1,290 @@
+                    data.Destination = (HANDLER_A == data.Destination) ? HANDLER_B : HANDLER_A;
+                    work.Identity = data.Destination;
+                    work.Datum = 0;
+                    for (var i = 0; i < Packet.DATA_SIZE; i++)
+                    {
+                        data.Count++;
+                        if (data.Count > 26)
+                        {
+                            data.Count = 1;
+                        }
+                        work.Data[i] = 65 + data.Count - 1;
+                    }
+                    return QueuePacket(work);
+                }
+            },
+        dataRecord);
+    }
+
+    public bool Start()
+    {
+        Packet workQ;
+
+        CreateIdler(IDLER, 0, NO_WORK, TaskState.CreateRunning());
+        workQ = CreatePacket(NO_WORK, WORKER, WORK_PACKET_KIND);
+        workQ = CreatePacket(workQ, WORKER, WORK_PACKET_KIND);
+
+        CreateWorker(WORKER, 1000, workQ, TaskState.CreateWaitingWithPacket());
+        workQ = CreatePacket(NO_WORK, DEVICE_A, DEVICE_PACKET_KIND);
+        workQ = CreatePacket(workQ, DEVICE_A, DEVICE_PACKET_KIND);
+        workQ = CreatePacket(workQ, DEVICE_A, DEVICE_PACKET_KIND);
+
+        CreateHandler(HANDLER_A, 2000, workQ, TaskState.CreateWaitingWithPacket());
+        workQ = CreatePacket(NO_WORK, DEVICE_B, DEVICE_PACKET_KIND);
+        workQ = CreatePacket(workQ, DEVICE_B, DEVICE_PACKET_KIND);
+        workQ = CreatePacket(workQ, DEVICE_B, DEVICE_PACKET_KIND);
+
+        CreateHandler(HANDLER_B, 3000, workQ, TaskState.CreateWaitingWithPacket());
+        CreateDevice(DEVICE_A, 4000, NO_WORK, TaskState.CreateWaiting());
+        CreateDevice(DEVICE_B, 5000, NO_WORK, TaskState.CreateWaiting());
+
+        Schedule();
+
+        return queuePacketCount == 23246 && holdCount == 9297;
+    }
+
+    internal TaskControlBlock FindTask(int identity)
+    {
+        TaskControlBlock t = taskTable[identity];
+        return NO_TASK == t ? throw new Exception("findTask failed") : t;
+    }
+
+    internal TaskControlBlock HoldSelf()
+    {
+        holdCount++;
+        currentTask!.IsTaskHolding = true;
+        return currentTask.Link;
+    }
+
+    internal TaskControlBlock QueuePacket(Packet packet)
+    {
+        Debug.Assert(currentTask != null);
+        TaskControlBlock t = FindTask(packet.Identity);
+        if (t == NO_TASK)
+        {
+            return NO_TASK;
+        }
+
+        queuePacketCount++;
+
+        packet.Link = NO_WORK;
+        packet.Identity = currentTaskIdentity;
+        return t.AddInputAndCheckPriority(packet, currentTask);
+    }
+
+    internal TaskControlBlock Release(int identity)
+    {
+        Debug.Assert(currentTask != null);
+        TaskControlBlock t = FindTask(identity);
+        if (NO_TASK == t)
+        {
+            return NO_TASK;
+        }
+        t.IsTaskHolding = false;
+        return t.Priority > currentTask.Priority ? t : currentTask;
+    }
+
+    [Conditional("TRACING")]
```

Comment:
Please don't do anything fancy like this.
Please stick to the basic core language.